src/WebJobs.Script.WebHost/Security/ISecretsRepository.cs

```diff
@@ -15,8 +15,10 @@ public interface ISecretsRepository
 
         Task<string> ReadAsync(ScriptSecretsType type, string functionName);
 
-        Task WriteAsync(ScriptSecretsType type, string functionName, string secretsContent);
+        Task WriteAsync(ScriptSecretsType type, string functionName, string secretsContent, bool isNonDecryptable = false);
```

Comment:
Based on our discussion, I thought we'd be enhancing the repository interface to introduce the notion of snapshots. We shouldn't be pushing anything related to encryption into API as the repository itself does't deal with encryption/decryption and shouldn't be concerned with that.